connectors/radon.py

```diff
@@ -0,0 +1,251 @@
+
+import logging
+import os
+
+from models.metric import Metric
+from radon.complexity import cc_visit, average_complexity
+from radon.raw import analyze, Module
+from radon.visitors import Function, Class
+from statistics import mean
+
+from utils.math import Math
+from utils.proglang import is_python_file
+
+
+class RadonConnector:
+    """
+    Class that allows calculating code metrics using the Radon library.
+    """
+
+    def __init__(self, directory, version, session, config) -> None:
+        """
+        Connector to Radon library to compute metrics.
+    
+        Attributes:
+        -----------
+        directory: str
+            Directory path containing source code files.
+        version: str
+            Version number of the code.
+        session: requests.Session
+            Session object to use to acces the bd.
+        config: Dict[str, Any]
+            Configuration options for the connector.
+        
+        Metrics:
+        --------
+        cc: List[float]
+            Cyclomatic complexity.
+        avg_cc: List[float]
+            Radon's average Cyclomatic complexity.
+        loc: List[int]
+            Total lines of code.
+        lloc: List[int]
+            Logical lines of code.
+        sloc: List[int]
+            Source lines of code.
+        comments: List[int]
+            Total number of comments.
+        docstring: List[int]
+            Total number of docstrings.
+        blank: List[int]
+            Total number of blank lines.
+        single_comment: List[int]
+            Total number of single-line comments.
+        noc: List[int]
+            Number of classes.
+        nom: List[int]
+            Number of methods.
+        nof: List[int]
+            Number of functions.
+        class_loc: List[int]
+            Lines of code for classes.
+        method_loc: List[int]
+            Lines of code for methods.
+        func_loc: List[int]
+            Lines of code for functions.
+        """
+        # Global config
+        self.directory = directory
+        self.version = version
+        self.session = session
+        self.config = config
+        # Metrics
+        self.cc = []                            
+        self.avg_cc = []                        
+        self.loc = []
+        self.lloc = []                          
+        self.sloc = []                          
+        self.comments = []                      
+        self.docstring = []                     
+        self.blank = []                         
+        self.single_comment = []                     
+        self.noc = []                           
+        self.nom = []                           
+        self.nof = []                           
+        self.class_loc = []                     
+        self.method_loc = []                    
+        self.func_loc = []        
```

Comment:
Too many attributes directly unrelated to the connector in my opinion. I think it would be better to use an intermediate Metric-like object, defined in this file (ex:` _RawMetric`) and have it as attribute of the connector `self.raw_metrics = _RawMetric()`